core/src/main/scala/kafka/coordinator/group/GroupCoordinatorAdapter.scala

```diff
@@ -62,6 +63,15 @@ private[group] class GroupCoordinatorAdapter(
   private val time: Time
 ) extends org.apache.kafka.coordinator.group.GroupCoordinator {
 
+  override def consumerGroupHeartbeat(
+    context: RequestContext,
+    request: ConsumerGroupHeartbeatRequestData
+  ): CompletableFuture[ConsumerGroupHeartbeatResponseData] = {
+    FutureUtils.failedFuture(Errors.UNSUPPORTED_VERSION.exception(
+      s"The old group coordinator does not support ${ApiKeys.CONSUMER_GROUP_HEARTBEAT.name} API."
```

Comment:
Does this error message make sense? I thought we already failed for the "old group coordinator" in KafkaApis, so wouldn't this run only if we were using the new coordinator?

Reply:
yeah... this code will never but hit if we keep the check in KafkaApis. i had to put something here just in case so it is better to actually put a meaningful message.

Reply:
Ah ok. I see. Makes sense and thanks for clarifying.